src/lib.zig

```diff
@@ -3015,6 +3033,251 @@ pub const Lua = struct {
+        const index = lua.absIndex(raw_index);
+        switch (@typeInfo(T)) {
+            .Int => {
+                switch (comptime lang) {
+                    .lua51, .luajit => {
+                        const result = lua.toInteger(index);
+                        return @as(T, @intCast(result));
+                    },
+                    else => {
+                        const result = try lua.toInteger(index);
+                        return @as(T, @intCast(result));
+                    },
+                }
+            },
+            .Float => {
+                switch (comptime lang) {
+                    .lua51, .luajit => {
+                        const result = lua.toNumber(index);
+                        return @as(T, @floatCast(result));
+                    },
+                    else => {
+                        const result = try lua.toNumber(index);
+                        return @as(T, @floatCast(result));
+                    },
+                }
+            },
+            .Pointer => |param_info| {
+                switch (param_info.size) {
+                    .Slice, .Many => {
+                        if (param_info.child != u8) {
+                            @compileError("Only u8 arrays (strings) may be parameters");
+                        }
+                        if (!param_info.is_const) {
+                            @compileError("Slice must be a const slice");
+                        }
+                        const string: [*:0]const u8 = try lua.toString(index);
+                        const end = std.mem.indexOfSentinel(u8, 0, string);
+
+                        if (param_info.sentinel == null) {
+                            return string[0..end];
+                        } else {
+                            return string[0..end :0];
+                        }
+                    },
+                    else => {
+                        return try lua.toUserdata(param_info.child, index);
+                    },
+                }
+            },
+            .Bool => {
+                return lua.toBoolean(index);
+            },
+            .Struct => {
+                return try lua.toStruct(T, index);
+            },
+            .Optional => {
+                if (lua.isNil(index)) {
+                    lua.pop(1);
+                    return null;
+                } else {
+                    return try lua.toAny(@typeInfo(T).Optional.child, index);
+                }
+            },
+            else => {
+                @compileError("Invalid parameter type");
+            },
+        }
+    }
+
+    /// Converts value at given index to a zig struct if possible
+    fn toStruct(lua: *Lua, comptime T: type, raw_index: i32) !T {
+        const index = lua.absIndex(raw_index);
+
+        if (!lua.isTable(index)) {
+            std.log.err("Parsing lua table failed because value at index {} is not a table", .{raw_index});
+            return error.value_not_a_table;
```

Comment:
Errors should follow the Zig style guide and use TitleCase

`error.NotATable` or something like that

Reply:
Oh I wasn't aware of that being a standard. I always write all my errors in snake_case. I can update that